### PR TITLE
Request code review to merge feature to dev

### PR DESCRIPTION
**Pull request by fedgut**
Request code review to merge enumerate methods feature to dev

### CODE REVIEW

File: scripts/enumerate.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  def my_any?
+    my_each { |i| return true if yield(i) }
+    false
+  end
+
+  def my_none?
+    my_each { |i| return false if yield(i) }
+    true
+  end
+
```
Comment (Ispirett) [inline scripts/enumerate.rb]: **These methods** should have a parameter.
without it I'm getting an ArgumentError: wrong number of arguments (given 1, expected 0)
should be able to handle when a block is not given.

File: scripts/enumerate.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    result = []
+    if block.nil?
+      my_each { |i| result.push(yield(i)) }
+    else
+      my_each { |i| result.push(block.call(i)) }
+    end
+    result
+  end
+
+  def my_inject(init = nil)
```
Comment (Ispirett) [inline scripts/enumerate.rb]: **My_inject** should be able to handle when a block is not given.
When testing with
  undefined local variable or method `length' for 5..50:Range

File: scripts/enumerate.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      i += 1
+      x = i
+    end
+  end
+
+  def my_select
+    result = []
+    my_each { |i| result.push(i) if yield(i) }
+    result
+  end
```
Comment (Ispirett) [inline scripts/enumerate.rb]: **These methods**
should be able to handle when a block is not given.

**Review by Ispirett — CHANGES_REQUESTED** [review]
Hey! @fedgut  great work so far :+1:  :100:  there are just some bugs :bug: to address below.

submit for review when done.

File: lib/enum.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  def my_all?(pattern = nil)
```
Comment (Oghenebrume50) [inline lib/enum.rb:28]: Should work with classes passed as arguments.  And work with regular pattern too

File: lib/enum.rb [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      my_each { |i| return false unless yield(i) }
+    elsif pattern.nil?
+      my_each { |i| return false unless i }
+    else
+      my_each { |i| return false unless i =~ pattern }
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
```
Comment (Oghenebrume50) [inline lib/enum.rb:39]: Should work with classes passed as arguments.  And work with regular pattern too

File: lib/enum.rb [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      my_each { |i| return true if yield(i) }
+    elsif pattern
+      my_each { |i| return true if i =~ pattern }
+    else
+      my_each { |i| return true if i }
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
```
Comment (Oghenebrume50) [inline lib/enum.rb:50]: Should work with classes passed as arguments.  And work with regular pattern too

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Hi @fedgut good Job :smile: there are just very few changes left to get this project approved, I left the comments on them. 
Please make changes and request another review. 
**[IMPORTANT]** It has been noticed that you and your partner aren't following the curriculum order in the sense that you are currently working on a project later down the line after this, please this is not ideal, the curriculum was created as a guide,  it is adviced to take this one step at a time, you guys will benefit more this way. :+1:

File: lib/enum.rb [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |elem| return false unless yield(elem) }
+    elsif pattern
+      my_each { |elem| return false unless verify?(elem, pattern) }
+    else
+      my_each { |elem| return false unless elem }
+    end
+    true
+  end
```
Comment (stratospherique) [inline lib/enum.rb:37]: * [x] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: lib/enum.rb [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  def my_any?(pattern = nil)
+    if block_given?
+      my_each { |elem| return true if yield(elem) }
+    elsif pattern
+      my_each { |elem| return true if verify?(elem, pattern) }
+    else
+      my_each { |elem| return true if elem }
+    end
+    false
+  end
```
Comment (stratospherique) [inline lib/enum.rb:48]: * [x] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: lib/enum.rb [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+  def my_none?(pattern = nil)
+    if block_given?
+      my_each { |elem| return false if yield(elem) }
+    elsif pattern
+      my_each { |elem| return false if verify?(elem, pattern) }
+    else
+      my_each { |elem| return false if elem }
+    end
+    true
+  end
```
Comment (stratospherique) [inline lib/enum.rb:59]: * [x] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: . However, some of your methods are not satisfying the requirements. check :arrow_down: 

- Check the code comments below :arrow_down: 

### That's it. When finished, kindly resubmit another code review form :pray: 

##### Ahmed Mahfoudh TSE

**Review by whiz25 — APPROVED** [review]
## Awesome :clap: 
You have made the requested changes and therefore it's a :+1:  from me. Please merge and proceed.

#### Martin Afani

**Pull request merged**